HydroModelBuilder/ModelInterface/flopyInterface/flopyInterface.py

```diff
@@ -391,27 +391,25 @@ def add_bc_to_target(self, bc_array, target):
         return target
     # End add_bc_to_river()
 
-    def cell_bc_to_3D_array_for_plot(self, boundary, bc_array, nper, use_final=False):
+    def cell_bc_to_3D_array_for_plot(self, boundary, bc_array, nper, val_pos):
         '''Convert cell data to 3D array for 3D plotting
-        
+
         :param boundary: str, name of boundary condition
-        :param bc_array: dict of list data for different boundaries including 
-                         cell location layer, row, column and other pertinent 
+        :param bc_array: dict of list data for different boundaries including
+                         cell location layer, row, column and other pertinent
                          data required by MODFLOW depending on the package
         :param nper: int, time period to use from bc_array
-        :param use_final: bool, force retrieval of last stress period data for that boundary
+        :param val_pos: int, target value position in bc_array
         '''
-        if use_final:
-            nper = max(bc_array.keys())
-        # End if
         self.bc_3D_array[boundary] = np.zeros_like(self.model_data.model_mesh3D[1])
```

Comment:
How crucial is this attribute? Do other methods rely on it?

I'm wondering if things can be simplified so that the method simply returns the generated array

Reply:
In essence, yes. "Generated array" was a poor choice of words. I was referring to "the array that is assigned to the object attribute and is updated inside this method".

I wasn't sure if you had some other workflow that relies on `bc_3D_array` being an attribute of the object, similar to what occurs in the `createNNNpackage()` methods, wherein MODFLOW package related methods only work if the specifically related attributes exist in later stages of the workflow.

If this is not the case, it would be simpler (and faster, use less memory) to just pass in the (pre-created) array and return the updated array.

Reply:
Actually looking at it again, it's not clear where `self.bc_3D_array` (which I just realised is actually a `dict` of arrays) is used/referred to so it's currently difficult to refactor.

Could you elaborate on the intended workflow? 
If it should be a variable that gets passed around where appropriate, then should `buildMODFLOW` return the dict/array?

Reply:
IT's just a dict I use to create some viz stuff and has no other purpose, so I'll make sure to shift it to flopy_viz